pkg/settlement/swap/transaction/transaction.go

```diff
@@ -56,12 +69,21 @@ func NewService(logger logging.Logger, backend Backend, signer crypto.Signer) (S
 		backend: backend,
 		signer:  signer,
 		sender:  senderAddress,
+		store:   store,
 	}, nil
 }
 
 // Send creates and signs a transaction based on the request and sends it.
 func (t *transactionService) Send(ctx context.Context, request *TxRequest) (txHash common.Hash, err error) {
-	tx, err := prepareTransaction(ctx, request, t.sender, t.backend)
+	t.lock.Lock()
```

Comment:
no, it's important the nonce is only increased after we actually broadcast the transaction so that if we were to retry afterward in case of failure (e.g. backend connection, signer issues) we use the same nonce again. also another call of `Send` might happen at the same time which would use the next nonce already and we risk the case where we broadcast a tx with nonce 6 but none of nonce 5 (then our tx would be stuck until the user manually makes a nonce 5 tx outside of bee).